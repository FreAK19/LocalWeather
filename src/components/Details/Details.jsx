import React from 'react';
import PropTypes from 'prop-types'
import './Details.less'

const Details = ({wind, pressure, humidity, min, max}) => (
 <div className='detail'>
   <table className='detail__table'>
     <tbody>
     <tr>
       <td>Wind Speed</td>
       <td>30 Km/h</td>
     </tr>
     <tr>
       <td>Humidity</td>
       <td>30 %</td>
     </tr>
     <tr>
       <td>Pressure</td>
       <td>30 hPa</td>
     </tr>
     <tr>
       <td>Max</td>
       <td>30 C</td>
     </tr>
     <tr>
       <td>Min</td>
       <td>30 C</td>
     </tr>
     </tbody>
   </table>
 </div>
);

Details.propTypes = {
  wind: PropTypes.number.isRequired
};

export default Details;
