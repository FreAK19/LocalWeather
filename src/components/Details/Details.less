@viewport-size-tablet: 640px;
@viewport-size-phone: 510px;

.detail {
  border-left: 4px solid rgb(38, 50, 56);
  flex-basis: 65%;

  &__table {
    padding: 1.875rem 3.121rem 2.75rem 3.735rem;
    display: block;
    text-align: left;
    font-weight: 400;

    & tr td {
      padding-right: 1rem;
    }
  }

  @media screen and (max-width: @viewport-size-tablet) {
    & {
      border-left: 4px solid rgb(209, 209, 209);
      flex-basis: 100%;
    }

    &__table {
      padding: 1.875rem 0 2.75rem 2.735rem;
    }
  }

  @media screen and (max-width: @viewport-size-phone) {
    & {
      border-left: 0;
    }
  }
}