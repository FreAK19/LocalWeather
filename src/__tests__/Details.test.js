import React from 'react';
import Details from '../components/Details/Details';

test('Details should render correctly', () => {
  const obj = {
    temp: 30,
    pressure: 100,
    humidity: 2330,
    temp_min: 30,
    temp_max: 35,
    speed: 10
  };
  const component = shallow(<Details details={obj} />);
  expect(component).toMatchSnapshot();
});
