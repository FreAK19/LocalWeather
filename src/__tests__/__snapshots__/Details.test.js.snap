// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should render correctly 1`] = `
<div
  className="detail"
>
  <table
    className="detail__table"
  >
    <tbody>
      <tr>
        <td>
          Wind Speed
        </td>
        <td>
          10
           m/sec
        </td>
      </tr>
      <tr>
        <td>
          Humidity
        </td>
        <td>
          2330
           %
        </td>
      </tr>
      <tr>
        <td>
          Pressure
        </td>
        <td>
          100
           hPa
        </td>
      </tr>
      <tr>
        <td>
          Max
        </td>
        <td>
          35
           C
        </td>
      </tr>
      <tr>
        <td>
          Min
        </td>
        <td>
          30
           C
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
